src/main/java/org/opensearch/neuralsearch/processor/NeuralQueryProcessor.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor;
+
+import java.util.Map;
+
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.ingest.ConfigurationUtils;
+import org.opensearch.neuralsearch.query.visitor.NeuralSearchQueryVisitor;
+import org.opensearch.search.pipeline.AbstractProcessor;
+import org.opensearch.search.pipeline.Processor;
+import org.opensearch.search.pipeline.SearchRequestProcessor;
+
+/**
+ * Neural Search Query Request Processor
+ */
+public class NeuralQueryProcessor extends AbstractProcessor implements SearchRequestProcessor {
+
+    /**
+     * Key to reference this processor type from a search pipeline.
+     */
+    public static final String TYPE = "neural_query";
```

Comment:
is this the agreed name for this processor? It sounds like a processor suppose to execute the query, maybe something describing actual functionality like enriching query with missing data.

Reply:
I personally think this current one looks good, we are anyway explaining everything in docs and other places. Type can  be a little lighter as well

Reply:
I would go with something like `enriching_query_defaults`, that is still light to me. The docs will be there for sure, what I mean is that having just `neural_query` gives wrong impression that we're executing that query, it's not about less details but about wrong direction. Plus with more generic name we can re-use that same processor for defaults in other fields.

Reply:
makes sense.

Reply:
@martin-gaievski Enriching Query Processor is too generic. Can we rename it to NeuralQueryEnrichingProcessor. In my understanding we should limit the scope of this processor to only neural queries. Nothing else. If we go too generic then this will become a default processor to put any kind of enriching which will bloat up the responsibilities of this processor.